apps/mobile/src/components/ChatDetailScreen/components/MessagesList.tsx

```diff
@@ -15,16 +25,29 @@ function renderItem({
 }
 
 function MessagesList(): JSX.Element {
-  const {messagesListAtomAtoms} = useMolecule(chatMolecule)
+  const {
+    messagesListAtomAtoms,
+    handleIsRevealIdentityOrContactRevealMessageVisibleActionAtom,
+  } = useMolecule(chatMolecule)
   const dataAtoms = useAtomValue(messagesListAtomAtoms)
+  const handleIsRevealIdentityOrContactRevealMessageVisible = useSetAtom(
+    handleIsRevealIdentityOrContactRevealMessageVisibleActionAtom
+  )
 
   return (
     <FlashList
       data={dataAtoms}
       estimatedItemSize={54}
+      contentContainerStyle={contentStyle.contentContainerStyle}
       keyExtractor={atomKeyExtractor}
       inverted
       renderItem={renderItem}
+      onViewableItemsChanged={
+        handleIsRevealIdentityOrContactRevealMessageVisible
+      }
+      viewabilityConfig={{
+        itemVisiblePercentThreshold: LIST_ITEM_VISIBILITY_PERCENTAGE_THRESHOLD,
```

Comment:
Move object creation out of react component just to be sure this does not cause any unnecessary rerenders